src/format/formatting.rs

```diff
@@ -611,38 +615,54 @@ pub(crate) fn write_rfc3339(
 }
 
 #[cfg(any(feature = "alloc", feature = "std"))]
-/// write datetimes like `Tue, 1 Jul 2003 10:52:37 +0200`, same as `%a, %d %b %Y %H:%M:%S %z`
+/// Write datetimes like `Tue, 1 Jul 2003 10:52:37 +0200`, similar to `%a, %d %b %Y %H:%M:%S %z`.
+///
+/// # Errors
+///
+/// RFC 2822 is only defined on years 0 through 9999, and returns an error on dates outside this
+/// range.
```

Comment:
Perhaps mention

    /// Uses `default_locale()` for locale information.

Reply:
I am a bit uncomfortable with the fact that `write_rfc2822_inner` localizes the day and month.
Even while using `format_localized` I think the `RFC2822` item should use English names as defined in the standard.

@djc Shall I remove localization from `write_rfc2822`? It lives behind the `unstable-locales` feature, so making a change in behaviour seems ok.

Reply:
> write_rfc2822_inner localizes the day and month.
Even while using format_localized I think the RFC2822 item should use English names as defined in the standard.

I agree: follow the standards.

Reply:
Agreed.